File: Industrial-Organization/exercise0417.m
% 2019/04/17
% Exercise in IO class
% Estimate the following equation
% y_i = \beta_0 + \beta_1^{x1_i} + \beta_2 * x2 + \epsilon


DATA = load('DataProblem201904.csv');  %Date, size=(1000,3)
y = DATA(:,1);
n = size(DATA, 1);
x0 = ones(n,1);
x1 = DATA(:,2);
x2 = DATA(:,3);

m = size(DATA,2);
theta = ones(m ,1);
f = @nlobj;
fminsearch(f, theta)

function objvalue = nlobj(theta)
global x0 x1 x2 y n;

sum = 0;
for i = 1:n
   sum = sum + (y(i) - theta(1)*x0(i) - theta(2)^x1(i) - theta(3)*x2(i))^2;
end
objvalue = sum;
end

% ans = 1, 2, 3

File: Industrial-Organization/exercise0417.py
"""
2019/04/17
Exercise in IO class
Python code

Estimate the following equation
y_i = \beta_0 + \beta_1^{x1_i} + \beta_2 * x2 + \epsilon
"""

import pandas as pd
import numpy as np
from scipy.optimize import fmin


def nlog(theta):
	data = pd.read_csv('DataProblem201904.csv', sep=',', header=None) #data
	y = np.asarray(data[0])
	x1 = np.asarray(data[1])
	x2 = np.asarray(data[2])
	n = len(y)

	total = 0
	for i in range(n):
		total += (y[i] - theta[0] - theta[1]**x1[i] - theta[2]*x2[i])**2
	return total

theta = np.ones(3) #initial values
fmin(nlog, theta)

"""
Result:

Optimization terminated successfully.
        Current function value: 0.000001
        Iterations: 80
        Function evaluations: 147
[0.9999538  2.00003254 3.00002048]
"""